hack/generator/pkg/codegen/storage/storage_type_factory.go

```diff
@@ -0,0 +1,465 @@
+	if err != nil {
+		return nil, err
+	}
+
+	p := prop.WithType(propertyType).
+		MakeOptional().
+		WithDescription("")
+
+	return p, nil
+}
+
+/*
+ * Functions used by the storageConverter TypeVisitor
+ */
+
+// convertResourceType creates a storage variation of a resource type
+func (f *StorageTypeFactory) convertResourceType(
+	tv *astmodel.TypeVisitor,
+	resource *astmodel.ResourceType,
+	ctx interface{}) (astmodel.Type, error) {
+
+	// storage resource types do not need defaulter interface, they have no webhooks
+	rsrc := resource.WithoutInterface(astmodel.DefaulterInterfaceName)
+
+	return astmodel.IdentityVisitOfResourceType(tv, rsrc, ctx)
+}
+
+// convertObjectType creates a storage variation of an object type
+func (f *StorageTypeFactory) convertObjectType(
+	_ *astmodel.TypeVisitor, object *astmodel.ObjectType, _ interface{}) (astmodel.Type, error) {
+
+	var errs []error
+	properties := object.Properties()
+	for i, prop := range properties {
+		p, err := f.makeStorageProperty(prop)
+		if err != nil {
+			errs = append(errs, errors.Wrapf(err, "property %s", prop.PropertyName()))
+		} else {
+			properties[i] = p
+		}
+	}
+
+	if len(errs) > 0 {
+		err := kerrors.NewAggregate(errs)
+		return nil, err
+	}
+
+	objectType := astmodel.NewObjectType().WithProperties(properties...)
+	return astmodel.StorageFlag.ApplyTo(objectType), nil
+}
+
+// redirectTypeNamesToStoragePackage modifies TypeNames to reference the current storage package
+func (f *StorageTypeFactory) redirectTypeNamesToStoragePackage(
+	_ *astmodel.TypeVisitor, name astmodel.TypeName, _ interface{}) (astmodel.Type, error) {
+	if result, ok := f.tryConvertToStorageNamespace(name); ok {
+		return result, nil
+	}
+
+	return name, nil
+}
+
+// stripAllValidations removes all validations
+func (f *StorageTypeFactory) stripAllValidations(
+	this *astmodel.TypeVisitor, v *astmodel.ValidatedType, ctx interface{}) (astmodel.Type, error) {
+	// strip all type validations from storage types,
+	// act as if they do not exist
+	return this.Visit(v.ElementType(), ctx)
+}
+
+// stripAllFlags removes all flags
+func (f *StorageTypeFactory) stripAllFlags(
+	tv *astmodel.TypeVisitor,
+	flaggedType *astmodel.FlaggedType,
+	ctx interface{}) (astmodel.Type, error) {
+	if flaggedType.HasFlag(astmodel.ARMFlag) {
+		// We don't want to do anything with ARM types
+		return flaggedType, nil
+	}
+
+	return astmodel.IdentityVisitOfFlaggedType(tv, flaggedType, ctx)
+}
+
+/*
+ * Functions used by the propertyConverter TypeVisitor
+ */
+
+// useBaseTypeForEnumerations replaces an enumeration with its underlying base type
+func (f *StorageTypeFactory) useBaseTypeForEnumerations(
+	tv *astmodel.TypeVisitor, et *astmodel.EnumType, ctx interface{}) (astmodel.Type, error) {
+	return tv.Visit(et.BaseType(), ctx)
+}
+
+// shortCircuitNamesOfSimpleTypes redirects TypeNames that reference resources or objects into our
+// storage namespace, and replaces TypeNames that point to simple types (enumerations or
```

Comment:
Is the main reason that you need to do this for things like validated types where they had an alias before but since you're dropped validations there's no need for an alias now?